server/core/src/dataObjOpr.cpp

```diff
@@ -347,7 +347,8 @@ getDataObjInfo(
         snprintf( dataObjInfo->objPath, MAX_NAME_LEN, "%s/%s", tmpCollName, tmpDataName );
         rstrcpy( dataObjInfo->rescHier, tmpHierString, MAX_NAME_LEN );
         irods::error ret = resc_mgr.hier_to_leaf_id( tmpHierString, dataObjInfo->rescId );
-        if( !ret.ok() ) {
+        if (!ret.ok()) {
+            std::free(dataObjInfo);
```

Comment:
Should this be `freeDataObjInfo`?

Reply:
Or maybe even `freeAllDataObjInfo`

Reply:
Seems reasonable. I'm not sure if there was a reason I used `std::free` here.

Will investigate.

Reply:
Done.